validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -180,19 +196,25 @@
   private final ExceptionMap exceptionMap;
   private final String generation;
   private final List<DeviceCredential> deviceCredentials = new ArrayList<>();
+  private final TreeMap<String, Object> siteMetadata;
 
   private String deviceNumId;
 
   private CloudDeviceSettings settings;
 
   LocalDevice(
       File siteDir, File devicesDir, String deviceId, Map<String, JsonSchema> schemas,
-      String generation) {
+      String generation, Metadata siteMetadata) {
```

Comment:
Note the public interface loads it well structured, then it's converted for internal use